alphacsc/utils/dictionary.py

```diff
@@ -19,6 +19,33 @@ def get_D(uv_hat, n_channels):
     return uv_hat[:, :n_channels, None] * uv_hat[:, None, n_channels:]
 
 
+def flip_uv(uv, n_channels):
+    """Modify uv to ensure that, for each atom, the corresponding temporal
+    pattern v peaks in the positive. If not the case, v is flip by multiplying
+    by -1.
+
+    Parameter
+    ---------
+    uv: array (n_atoms, n_channels + n_times_atom)
+    n_channels: int
+        number of channels in the original multivariate series
+
+    Return
+    ------
+    uv: array (n_atoms, n_channels + n_times_atom)
+    """
+    v = uv[:, n_channels:]
+    index_array = np.argmax(np.absolute(v), axis=1)
+    val_index = np.take_along_axis(v, np.expand_dims(
+        index_array, axis=-1), axis=-1).squeeze(axis=-1)
+    # Get atoms indices that need to have their v flipped
+    index_to_flip = np.where(val_index < 0)[0]
+    v[index_to_flip] *= -1
+    # Update uv
+    uv[:, n_channels:] = v
```

Comment:
You also need to flip `u` to get the invariance. You can flip `v` because you are only interested in `uv^T` which is equal to `(-u)(-v)^T`.
```suggestion
    index_to_flip = np.where(val_index < 0)[0]
    uv[index_to_flip] *= -1
```